src/pubd/repository.rs

```diff
@@ -1030,30 +1032,37 @@ impl RepositoryAccessProxy {
         &self,
         rfc8181_uri: uri::Https,
         publisher: &PublisherHandle,
-    ) -> KrillResult<rfc8183::RepositoryResponse> {
+    ) -> KrillResult<idexchange::RepositoryResponse> {
         self.read()?.repository_response(rfc8181_uri, publisher)
     }
 
     /// Parse submitted bytes by a Publisher as an RFC8181 ProtocolCms object, and validates it.
-    pub fn validate(&self, publisher: &PublisherHandle, msg: Bytes) -> KrillResult<ProtocolCms> {
+    pub fn decode_and_validate(
+        &self,
+        publisher: &PublisherHandle,
+        bytes: &[u8],
+    ) -> KrillResult<publication::PublicationCms> {
         let publisher = self.get_publisher(publisher)?;
-        let msg = ProtocolCms::decode(msg, false).map_err(|e| Error::Rfc8181Decode(e.to_string()))?;
-        msg.validate(publisher.id_cert()).map_err(Error::Rfc8181Validation)?;
+        let msg = PublicationCms::decode(bytes).map_err(Error::Rfc8181)?;
+        msg.validate(publisher.id_cert()).map_err(Error::Rfc8181)?;
```

Comment:
Why was this changed from `Error::Rfc8181Validation` to `Error::Rfc8181` given that the context is validation?